### PR TITLE
Adds some procs that are useful for bus

### PR DESCRIPTION
**Pull request by adamsong**
# Github documenting your Pull Request

Funny bus

Darkspawn ascend, but its moja
Chestburst, but its moja

### CODE REVIEW

**Comment by alexkar598** [comment]
Going to get real boring real fast

File: yogstation/code/modules/admin/moja.dm [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(src.loc)
+		moja.ckey = src.ckey
+		qdel(src)
+
+/mob/living/proc/moja_burst(new_key)
+	var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(src.loc)
+	moja.ckey = (new_key ? new_key : ckey)
+	gib()
+
+/mob/living/carbon/human/moja
```
Comment (Bibby0110) [inline yogstation/code/modules/admin/moja.dm]: don't just have an empty type declaration. you don't need it, defining a proc on the type declares the type automatically

File: yogstation/code/modules/admin/moja.dm [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	if(istype(src,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = src
+		H.make_moja()
+	else
+		var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(src.loc)
+		moja.ckey = src.ckey
+		qdel(src)
+
+/mob/living/proc/moja_burst(new_key)
+	var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(src.loc)
```
Comment (Bibby0110) [inline yogstation/code/modules/admin/moja.dm:58]:
```suggestion
	var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(get_turf(loc))
```

File: yogstation/code/modules/admin/moja.dm [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	for(var/mob/living/L in view(7, src))
+		if(L == src)
+			continue
+		L.flash_act(1, 1)
+		L.Knockdown(50)
+	if(istype(src,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = src
+		H.make_moja()
+	else
+		var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(src.loc)
```
Comment (Bibby0110) [inline yogstation/code/modules/admin/moja.dm:53]:
```suggestion
		var/mob/living/carbon/human/moja/moja = new /mob/living/carbon/human/moja(get_turf(loc))
```

File: yogstation/code/modules/admin/moja.dm [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	animate(src, pixel_y = src.pixel_y + 8, time = 60)
+	sleep(45)
+	Shake(5, 5, 110)
+	for(var/i in 1 to 20)
+		to_chat(src, "<span class='velvet bold'>[pick("I- I- I-", "Mind-", "Sigils-", "Can't think-", "<i>POWER-</i>","<i>TAKE-</i>", "M-M-MOOORE-")]</span>")
+		sleep(1.1) //Spooky flavor message spam
+	visible_message("<span class='userdanger'>A tremendous shockwave emanates from [src]!</span>", "<span class='velvet big'><b>YOU ARE FREE!!</b></span>")
+	playsound(src, 'yogstation/sound/magic/divulge_end.ogg', 50, 0)
+	animate(src, color = initial(color), pixel_y = initial(pixel_y), time = 30)
+	for(var/mob/living/L in view(7, src))
```
Comment (Bibby0110) [inline yogstation/code/modules/admin/moja.dm:44]: could do oview and drop the self check at the expense of not stunning people standing on top of the moja

File: yogstation/code/modules/admin/moja.dm [hunk 5]
```diff
@@ -0,0 +1,87 @@
+/mob/living/proc/moja_divulge()
+	visible_message("<span class='warning'>A vortex of violet energies surrounds [src]!</span>", "<span class='velvet'>Your barrier will keep you shielded to a point..</span>")
+	visible_message("<span class='danger'>[src] slowly rises into the air, their belongings falling away, and begins to shimmer...</span>", \
+						"<span class='velvet big'><b>You begin the removal of your human disguise. You will be completely vulnerable during this time.</b></span>")
+	setDir(SOUTH)
+	for(var/obj/item/I in src)
+		dropItemToGround(I)
+	for(var/turf/T in RANGE_TURFS(1, src))
+		new/obj/structure/psionic_barrier(T, 500)
```
Comment (Bibby0110) [inline yogstation/code/modules/admin/moja.dm]:
```suggestion
		new/obj/structure/psionic_barrier(T)
```
default value

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
All the times need changing to use the defines
New procs need autodoccing

**Comment by Bibby0110** [comment]
personally think it's a weird bloaty way to include a moja reference

**Comment by SomeguyManperson** [comment]
unfunny and bluepilled

File: yogstation/code/modules/admin/moja.dm (hunk 5 above)
Comment (adamsong) [reply yogstation/code/modules/admin/moja.dm]: Copied from darkspawn, will fix

File: yogstation/code/modules/admin/moja.dm (hunk 4 above)
Comment (adamsong) [reply yogstation/code/modules/admin/moja.dm:44]: Moja might stand on someone though

File: yogstation/code/modules/admin/moja.dm (hunk 3 above)
Comment (adamsong) [reply yogstation/code/modules/admin/moja.dm:53]: I want them to show up in a locker if they are in a locker

File: yogstation/code/modules/admin/moja.dm (hunk 2 above)
Comment (adamsong) [reply yogstation/code/modules/admin/moja.dm:58]: I want them to show up in a locker if they are in a locker

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**